api/configuration/v1alpha1/kongsni_types.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2024 Kong, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	konnectv1alpha1 "github.com/kong/kubernetes-configuration/api/konnect/v1alpha1"
+)
+
+// KongSNI is the schema for SNI API which defines a Kong SNI.
+//
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Programmed",description="The Resource is Programmed on Konnect",type=string,JSONPath=`.status.conditions[?(@.type=='Programmed')].status`
+// +kubebuilder:validation:XValidation:rule="oldSelf.spec.certificateRef == self.spec.certificateRef", message="spec.certificateRef is immutable"
+type KongSNI struct {
```

Comment:
While there is an API that allows creating SNIs not attached to a certificate https://kong-platform-api.netlify.app/konnect/control-planes-config/v2/openapi.yaml/#tag/SNIs/operation/create-sni when using Konnect UI it's impossible to do so (see the screenshot below) and one has to specify the related certificate (and then we'd use https://kong-platform-api.netlify.app/konnect/control-planes-config/v2/openapi.yaml/#tag/SNIs/operation/create-sni-with-certificate)

<img width="825" alt="image" src="https://github.com/user-attachments/assets/3f15e48a-2d70-4248-9a84-8c1412d086c6">

Because of that I believe we should add the certificate ref and require it similarly how we do it for e.g. CP refs in other types.

WDYT?

Reply:
Added a `size(self.spec.certificateRef.name) > 0` rule to limit the `certificateRef.name` to be non-empty. I am not sure if this one or `...!=''` is better.